app/scripts/controllers/transactions/tx-state-manager.js

```diff
@@ -167,7 +167,12 @@ class TransactionStateManager extends EventEmitter {
         transactions.splice(index, 1)
       }
     }
-    transactions.push(txMeta)
+    const newTxIndex = transactions.findIndex((metaTx) => {
+      return metaTx.id === txMeta.id + 1
```

Comment:
It seems that this only works if the transaction is off-by-one, e.g. if the transaction that came in first was _two_ behind somehow, it would still be out of order.

Reply:
An elegant solution to this might be using the transaction id itself as the array index, and letting `transactions` be a _sparse_ array. It'd be a bit tangly though, as you'd need to ensure nobody every tried traversing it using the index (e.g. a `for` loop) and we'd need to update the serialization so that we don't save a million `null`'s to disk.

Reply:
Or we could use a similar approach to what you've done here, except accounting for differences of more than one.

e.g.
```
const newTxIndex = transactions.findIndex((currentTxMeta) => {
      return currentTxMeta.id > txMeta.id
})
```
That'd get the index of the _first_ transaction with a higher id, and let you place this one before it.

Reply:
This also has me wondering... is this something we want to do, generally?

The order of transactions in this list already doesn't match the order of `id`s, because of speedup/retry transactions (at least if I'm following how this controller works). Do we want to change where speedups/retry transactions appear in this list? 🤔 Perhaps the change to `transactions/index.js` is sufficient to prevent the test failure you were seeing?

Reply:
added your suggested change in ccd7e6257

Reply:
Thanks!

To follow up on the question of speedup/retry transactions, we since discovered that speedups do get assigned new ids (of course - not sure why I thought otherwise 🤦‍♂ ), so this change shouldn't affect those.